src/test/java/buisness/LoginPage.java

```diff
@@ -0,0 +1,40 @@
+package buisness;
+
+import com.codeborne.selenide.Selenide;
+import core.dto.LoginUserDto;
+import core.enums.UsersRole;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.codeborne.selenide.Condition.enabled;
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$x;
+import static core.enums.PropertyNames.BASE_URL;
+import static core.utils.services.ConfigurationService.getProperty;
+
+
+@Slf4j
+public class LoginPage extends BasePage {
+  private static final String LOGIN_FIELD_XPATH = "//input[@name='login']";
+  private static final String PASSWORD_FIELD_XPATH = "//input[@name='password']";
+  private static final String LOGIN_BUTTON_XPATH = "//button[@type='submit']";
+
+  private void openLoginPage() {
+    Selenide.open(getProperty(BASE_URL.getName()));
+  }
+
+  public void loginWithUser(UsersRole userRole) {
```

Comment:
I suggest leaving here only the simple logic of working with web elements and extracting the complex login logic to the service layer. The service layer is a middle layer between page layer and test layer where you can leave a complex business action with pages like login flow, adding new filters and e.t.c

Reply:
Yes, your right, I've done it for Login page as for now it is overloaded with not related to pageObjects methods